### ISSUE TITLE
Change Pod Hostname

### ISSUE THREAD

**Issue by 6543** [diaspora/diaspora#7665]
Is there a oportunety to change the hostname / dns-name of a pod corectly?
or is the only way to setup a new on and migrate the acounts ?

and if you have no time to run the old one for migratin time?

**Comment by SuperTux88** [comment]
No, currently you can't change the hostname. The only way you can do it is to setup a new pod and register new accounts (importing (parts of) the old database isn't supported either). It maybe will be possible in the future to rename the hostname using the account migration feature (#908), but that isn't finished yet.

I keep this issue open to track the hostname-rename-functionality once the migration-feature is done.

**Comment by cmrd-senya** [comment]
Technically now, as the `AccountMigration` message is supported on the federation level it is possible to change pod hostname by creating the `AccountMigration` object and performing the migration for each pod user. Though this is something to be tested.

**Comment by goobertron** [comment]
@cmrd-senya would it make sense to set up a small production pod, make some accounts on it, get some federation going and then change the domain and see if you're still federating properly?

**Comment by cmrd-senya** [comment]
It would make sense, but I don't know how important is it. It would definitely help us as a way to test the migration code, but we will do the testing anyway and testing this specific case is more important if someone is going to use it in production. While there is no one who wants and is going to change the hostname of theirs pod, this is not a priority, IMO.

**Comment by 6543** [comment]
I will switch to a shorter name in the future, by now how can i test it on a extra pod?

**Comment by goobertron** [comment]
>  by now how can i test it on a extra pod?

I suggest setting up a fresh production pod and creating some test accounts on it, share with other people you know on other pods, perhaps including your own accounts on other pods, and then go through the process described above to migrate to a new domain. To save buying extra domains, you could probably check it by creating new subdomains for an existing domain. Then see if your accounts, now on this new (sub)domain, are still federating with other pods. @cmrd-senya would need to confirm whether this method would properly test this feature, but that's what seems to me to be what is needed.

**Comment by Flaburgan** [comment]
>share with other people you know on other pods, perhaps including your own accounts on other pods

This should probably limited only to pods with podmins involved here imo, as if something goes **really** wrong (who knows?) some cleaning may be required on the other pods too.

**Comment by SuperTux88** [comment]
You can just test this with two development pods (you need to disable `single_process_mode` or increase the number of unicorn workers to make federation work in development mode), you don't need to create an extra pod in the production network for that, which would then leave two dead pods in the pods list in the network (first the extra pod and second the renamed extra pod which will be dead too afterwards, because when it's an extra pod for that it's only temporary).

I'm also not sure about making this an official feature, because it would add a lot of load to all pods when a big pod would change the hostname (small/single-user pods are no problem). But we can't prevent users from using the account-migration for that or simply send mass-migrations, so maybe we just should add a protection for that, so it would just take very long when a big pod changes the hostname, but it wouldn't block other federation stuff. Maybe we could do it similar as it's already done to prevent parallel exports [here](https://github.com/diaspora/diaspora/blob/develop/app/workers/export_user.rb#L14), or as we use this for more workers, we can maybe start using something like [sidekiq-limit_fetch](https://github.com/brainopia/sidekiq-limit_fetch), because the way it's done for the exports is more or less only a "hack". The easiest way to do something like this would be to just start multiple sidekiq processes for different queues with different concurrencies, but that would then use more memory (which would be a problem for smaller pods).

**Comment by r3pek** [comment]
I don't mind volunteering to test the feature since I'm gonna change the Diaspora ID domain, but not the server domain name. If I can avoid a instance reset, it would be great.
